codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -236,7 +236,8 @@ private void writePendingBytes() {
 
     /**
      * This will allocate bytes by stream weight and priority for the entire tree rooted at {@code parent}, but does
-     * not write any bytes.
+     * not write any bytes. Priority is generally distributed proportionally to siblings in the priority tree, however
+     * we ensure that at least 1 byte is assigned to each stream in order to make progress.
```

Comment:
@Scottmitch any thoughts on this?

I'm tempted to get rid of `max(1,...` altogether and let the next iteration deal with it. This could, however, result in stream starvation in very extreme cases although I'm not sure how often this would happen in practice.  One way that we might deal with it is by keeping some hysteresis on allocation to artificially bump up the weight for streams that haven't been allocated any bytes in a while.